src/proto/streams/streams.rs

```diff
@@ -312,7 +312,13 @@ where
 impl<B> DynStreams<'_, B> {
     pub fn recv_headers(&mut self, frame: frame::Headers) -> Result<(), Error> {
         let mut me = self.inner.lock().unwrap();
-
+        if me.store.num_wired_streams() > me.counts.max_streams() {
+            tracing::error!("HEADERS: number of streams exceeds the upper limit ({})", me.counts.max_streams());
+            return Err(Error::remote_go_away(
```

Comment:
Its not clear to me that this is the correct response as stated above. Would be good to get some reference from the spec or any other implementations for what they do here.

Reply:
I didn't find anything more useful
https://www.rfc-editor.org/rfc/rfc7540#section-6.5.2